### PR TITLE
Increase build performance w/ Gradle upgrade to 7.4

### PR DESCRIPTION
**Pull request by am357**
*Issue #, if available:* 538

*Description of changes:*
See the attached issue for motivation.

This commit contains the following changes:
1. Required settings for parallelizing test executions for each sub-project.
2. Changes for removing blocking deprecated Gradle settings for upgrading Gradle to 7.4
3. Gradle 7.4 upgrade:
    In addition to other fixes, to be able to use features  like _File System Watching_ by default:
    https://docs.gradle.org/current/userguide/performance.html#file_system_watching
5. Enable parallel Gradle build and build cache:
    https://docs.gradle.org/current/userguide/build_cache.html
6. Plugin upgrade:
    As part of upgrade removing the dependency to jcenter repository.

With these changes, apart from parallelizing builds and test executions, our major gain is in builds after the first build.  This is because as appose to the current situation, the subsequent build times are dependent to the amount of incremental changes we make between each build.  For example in the following, the second build is after making no changes to the first build:

```bash
➜  partiql-lang-kotlin git:(improve-build-performance) ./gradlew build

> Task :lang:generatePigDomains
pig: universe file: /Users/maymandi/workplace/maymandi/partiql-lang-kotlin/lang/resources/org/partiql/type-domains/partiql.ion
pig: output file  : /Users/maymandi/workplace/maymandi/partiql-lang-kotlin/lang/src/org/partiql/lang/domains/partiql-domains.kt
pig: parsing the universe...
pig: permuting domains...
pig: applying Kotlin pre-processing
pig: applying the Kotlin template...
pig: universe generation complete!

> Task :lang:compileTestKotlin
...
Daemon will be stopped at the end of the build after running out of JVM memory
...
BUILD SUCCESSFUL in 2m 38s
49 actionable tasks: 27 executed, 22 up-to-date

➜  partiql-lang-kotlin git:(improve-build-performance) ./gradlew build
...

BUILD SUCCESSFUL in 10s
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -1,7 +1,7 @@
-#!/usr/bin/env sh
```
Comment (am357) [inline gradlew:1]: The changes in this file are auto-generated by Gradle 7.4 upgrade.

File: cli/build.gradle [hunk 2]
```diff
@@ -35,9 +34,9 @@ dependencies {
 
 jacocoTestReport {
     reports {
-        xml.enabled true
-        html.enabled true
-        csv.enabled false
-        html.destination file("${buildDir}/reports/jacoco/html")
+        xml.required.set(false)
+        html.required.set(true)
+        xml.required.set(false)
```
Comment (jpschorr) [inline cli/build.gradle:39]: lines 37 & 39 are duplicates

File: examples/build.gradle [hunk 3]
```diff
@@ -34,9 +33,9 @@ dependencies {
 
 jacocoTestReport {
     reports {
-        xml.enabled true
-        html.enabled true
-        csv.enabled false
-        html.destination file("${buildDir}/reports/jacoco/html")
+        xml.required.set(false)
+        html.required.set(true)
+        xml.required.set(false)
```
Comment (jpschorr) [inline examples/build.gradle:38]: 36 & 38 are duplicate lines

File: lang/build.gradle [hunk 4]
```diff
@@ -168,9 +170,9 @@ signing {
 
 jacocoTestReport {
     reports {
-        xml.enabled true
-        html.enabled true
-        csv.enabled false
-        html.destination file("${buildDir}/reports/jacoco/html")
+        xml.required.set(false)
+        html.required.set(true)
+        xml.required.set(false)
```
Comment (jpschorr) [inline lang/build.gradle:175]: 173, 175 duplicate lines

File: testscript/build.gradle [hunk 5]
```diff
@@ -28,9 +24,9 @@ compileTestKotlin {
 
 jacocoTestReport {
     reports {
-        xml.enabled true
-        html.enabled true
-        csv.enabled false
+        xml.required.set(false)
+        html.required.set(true)
+        xml.required.set(false)
```
Comment (jpschorr) [inline testscript/build.gradle:29]: 27 & 29 duplicates

File: pts/build.gradle [hunk 6]
```diff
@@ -41,9 +37,9 @@ dependencies {
 
 jacocoTestReport {
     reports {
-        xml.enabled true
-        html.enabled true
-        csv.enabled false
-        html.destination file("${buildDir}/reports/jacoco/html")
+        xml.required.set(false)
+        html.required.set(true)
+        xml.required.set(false)
```
Comment (jpschorr) [inline pts/build.gradle:42]: same duplication

**Review by jpschorr — COMMENTED** [review]
Several `build.gradle` files have:

```
        xml.required.set(false)
        html.required.set(true)
        xml.required.set(false)
```

Where the two `xml...` lines seem to be duplicates.

File: pts/build.gradle (hunk 6 above)
Comment (am357) [reply pts/build.gradle:42]: Nice catch! I think it makes sense to move them all in the main `build.gradle` and remove the duplicate altogether.
Will make the change in the next commit.

File: cli/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (alancai98) [reply cli/build.gradle:39]: Feel free to remove actually. Don't think I've ever used the CSV jacoco report before

**Review by jpschorr — APPROVED** [review]

**Review by alancai98 — APPROVED** [review]
:shipit:

**Pull request merged**